File: gas-station/gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int total_sum=0;
        int curr_sum=0;
        int start_index=0;

        for(int i=0;i<gas.size();i++)
        {
            total_sum+=(gas[i]-cost[i]);
            curr_sum+=(gas[i]-cost[i]);
            if(curr_sum<0)
            {
                curr_sum=0;
                start_index=i+1;
            }
        }
        return (total_sum<0)?-1:start_index;
    }
};

File: 134-gas-station/gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int n = gas.length;
        int surplus = 0;
        int totalsurplus = 0;
        int start =0;
        for(int i=0;i<n;i++)
        {
            totalsurplus += gas[i] - cost[i];
            surplus += gas[i] - cost[i];
            if(surplus<0)
            {
                surplus = 0;
                start = i+1;
            }
        }

        return totalsurplus<0 ? -1 : start;
    }
}